WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/logic/interval/intervaltypes/IntervalBase.java

```diff
@@ -122,7 +122,11 @@ public IntervalType getType() {
 
 	/** Necessary for storage of Intervals. */
 	public int compareTo(IntervalBase comparedInterval) {
-		return getEnd().compareTo(comparedInterval.getEnd());
+		int res = getEnd().compareTo(comparedInterval.getEnd());
+		if (res == 0 && !this.equals(comparedInterval)) {
+			res = -1;
+		}
+		return res;
```

Comment:
Hm, is this semantically correct?
Because assume two IntervalBase a, b and a != b.
Then compareTo(a,b) = -1
but compareTo(b, a) = -1 (and it should be compareTo(b,a) = 1).

Or am I missing something?